### PR TITLE
Add bytes support to ZMQ

### PR DESCRIPTION
**Pull request by e-kayrakli**
This PR adds support for the `bytes` type in the ZMQ module.

Summary:
- Mimic the support for send/recv `string`s.
- Update docs
- Update a ZMQ test to send/recv `bytes`

This is should ideally be merged before the UTF8 validation PR (#14384), so that
arbitrary data can be send/recv'd using `bytes` object with ZMQ.

Test status:
- [x] standard `test/library/packages/ZMQ` (run locally)
- [x] full standard
- [x] full gasnet

### CODE REVIEW

File: test/library/packages/ZMQ/pair.chpl [hunk 1]
```diff
@@ -121,6 +128,14 @@ proc Worker() {
     writeln("val = ", val);
   }
 
+  // Bytes
+  {
+    var val = b"The quick brown fox...";
```
Comment (vasslitvinov) [inline test/library/packages/ZMQ/pair.chpl]: What if you throw a non-UTF8 byte into this literal?

Comment (e-kayrakli) [reply test/library/packages/ZMQ/pair.chpl]: Good question.. I hope it works :) I'll make that change to the test

File: modules/packages/ZMQ.chpl [hunk 2]
```diff
@@ -167,12 +167,12 @@ Serialization
 +++++++++++++
 
 In Chapel, sending or receiving messages is supported for a variety of types.
-Primitive numeric types and strings are supported as the foundation.
-In addition, user-defined ``record`` types may be serialized automatically
-as `multipart messages <http://zguide.zeromq.org/page:all#Multipart-Messages>`_
-by internal use of the :chpl:mod:`Reflection` module.
-Currently, the ZMQ module can serialize records of primitive numeric types,
-strings, and other serializable records.
+Primitive numeric types, along with ``string`` and ``bytes`` are supported as the
```
Comment (vasslitvinov) [inline modules/packages/ZMQ.chpl]: Need a comma after `''bytes''`. Or, you can go simple and say "... types, ``strings``, and ``bytes`` are...", and keep the rest of the paragraph unchanged - which IMO looks better than with your change.

Comment (e-kayrakli) [reply modules/packages/ZMQ.chpl]: Yeah, I didn't like the second time I read as well. I'll keep the original and just add bytes to the list.

**Pull request merged**